ouroboros-consensus/src/Ouroboros/Consensus/Node/Tracers.hs

```diff
@@ -97,7 +100,7 @@ showTracers tr = Tracers
 -------------------------------------------------------------------------------}
 
 -- | Trace the forging of a block as a slot leader.
-data TraceForgeEvent blk
+data TraceForgeEvent blk txs
```

Comment:
Remove the `txs` parameter, we know the type of the transactions: `[GenTx blk]`.

Reply:
And how do I derive the `Eq` and `Show` for that?
Isn't that still abstract at this point and it is hard to derive?

Reply:
```haskell
deriving instance (ProtocolLedgerView blk, Eq blk, Eq (GenTx blk))
               => Eq   (TraceForgeEvent blk)
deriving instance (ProtocolLedgerView blk, Show blk, Show (GenTx blk))
               => Show (TraceForgeEvent blk)
```
That requires `StandaloneDeriving` and `UndecidableInstances`.

*However,* I realise now that there is some precedent in the tracers (`txInboundTracer`/`txOutboundTracer`) already. Can you just change the type parameter to `tx` instead of `txs`? You can wrap `tx` in a list inside of `TraceForgeEvent`.

Reply:
Will fix.